rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/DisconnectionRouter.java

```diff
@@ -63,31 +56,39 @@ public Boolean call(Boolean isAdapterUsable) {
     }
 
     /**
-     * Method to be called whenever a connection braking exception happens. It will be routed to {@link #asObservable()}.
-     *
-     * @param disconnectedException the exception that happened
+     * @inheritDoc
+     */
+    @Override
+    public void onDisconnectedException(BleDisconnectedException disconnectedException) {
+        disconnectionErrorInputRelay.call(disconnectedException);
+    }
+
+    /**
+     * @inheritDoc
      */
-    void onDisconnectedException(BleDisconnectedException disconnectedException) {
-        disconnectionErrorRelay.call(disconnectedException);
+    @Override
+    public void onGattConnectionStateException(BleGattException disconnectedGattException) {
+        disconnectionErrorInputRelay.call(disconnectedGattException);
     }
 
     /**
-     * Method to be called whenever a BluetoothGattCallback.onConnectionStateChange() will get called with status != GATT_SUCCESS
-     *
-     * @param disconnectedGattException the exception that happened
+     * @inheritDoc
      */
-    void onGattConnectionStateException(BleGattException disconnectedGattException) {
-        disconnectionErrorRelay.call(disconnectedGattException);
+    @Override
+    public Observable<BleException> asExactObservable() {
```

Comment:
What does it mean 'asExactObservable'? :) 😃  Could the method say something about the behavior?